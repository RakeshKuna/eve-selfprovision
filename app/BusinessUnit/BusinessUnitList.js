import React from 'react';
import { Row, Col, Button } from 'reactstrap';
import { ETable } from 'containers/ETable';
import { Link } from 'react-router-dom';
import userService from 'services';
import { alertActions } from 'actions';
import { store } from '../store';

export class BusinessUnitList extends React.Component {
  constructor() {
    super();

    this.state = {
      data: [],
      loading: false
    };

    this.columns = [{
      Header: 'Channel Partner Id',
      accessor: 'channelPartnerId' // String-based value accessors!
    }, {
      Header: 'Channel Partner Name',
      accessor: 'channelPartnerName' // String-based value accessors!
    }, {
      Header: 'Channel Partner Type',
      accessor: 'channelPartnerType'
    }, {
      Header: 'Currency',
      accessor: 'currency'
    }, {
      Header: 'Actions',
      accessor: '[row identifier to be passed to button]',
      Cell: (row) => (<Button color="primary" className="view_btn" tag={Link} to={`/editBu/${row.original.channelPartnerId}`}>View</Button>),
      filterable: false
    }];
  }

  componentDidMount() {
    this.handleLoad(true);
    userService.getBuList()
      .then(
        (success) => {
          if (Array.isArray(success.channelPartenrs)) {
            this.setState({
              data: success.channelPartenrs
            });
          } else {
            this.setState({
              data: [success.channelPartenrs]
            });
          }
          this.handleLoad(false);
        },
        (error) => {
          store.dispatch(alertActions.error(error));
          this.handleLoad(false);
        }
      );
  }

  // editBu(value) {
  //   // console.log(value);
  // }
 handleLoad=(loading) => {
   if (loading) {
     this.setState({
       loading: true
     });
   } else {
     this.setState({
       loading: false
     });
   }
 }
 render() {
   return (
     <div>
       <Row className="mb-4">
         <Col sm={8}>
           <h2>Business Unit List</h2>
         </Col>
         <Col sm={4} className="text-right">
           <Button color="primary" tag={Link} to="/addBU">Create BU</Button>
         </Col>
       </Row>
       <ETable key="A" data={this.state.data} columns={this.columns} loading={this.state.loading} />
     </div>
   );
 }
}
export default BusinessUnitList;
