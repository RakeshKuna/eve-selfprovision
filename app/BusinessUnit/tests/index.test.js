import React from 'react';
import { shallow } from 'enzyme';
import { BusinessUnitList } from '../BusinessUnitList';

describe('BusinessUnitList', () => {
  it('<Row> element should render', () => {
    const wrapper = shallow(<BusinessUnitList />);
    expect(wrapper).toMatchSnapshot();
  });
  it('<Row> on Button redirect', () => {
    const wrapper = shallow(<BusinessUnitList />);
    wrapper.find('Button').at(0).props('to');
  });
});
