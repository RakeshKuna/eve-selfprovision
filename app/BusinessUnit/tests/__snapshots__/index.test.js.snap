// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BusinessUnitList <Row> element should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BusinessUnitList />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Row
          className="mb-4"
          tag="div"
        >
          <Col
            sm={8}
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <h2>
              Business Unit List
            </h2>
          </Col>
          <Col
            className="text-right"
            sm={4}
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <Button
              color="primary"
              tag={[Function]}
              to="/addBU"
            >
              Create BU
            </Button>
          </Col>
        </Row>,
        <Connect(ETable)
          columns={
            Array [
              Object {
                "Header": "Channel Partner Id",
                "accessor": "channelPartnerId",
              },
              Object {
                "Header": "Channel Partner Name",
                "accessor": "channelPartnerName",
              },
              Object {
                "Header": "Channel Partner Type",
                "accessor": "channelPartnerType",
              },
              Object {
                "Header": "Currency",
                "accessor": "currency",
              },
              Object {
                "Cell": [Function],
                "Header": "Actions",
                "accessor": "[row identifier to be passed to button]",
                "filterable": false,
              },
            ]
          }
          data={Array []}
          loading={true}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Col
              sm={8}
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <h2>
                Business Unit List
              </h2>
            </Col>,
            <Col
              className="text-right"
              sm={4}
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <Button
                color="primary"
                tag={[Function]}
                to="/addBU"
              >
                Create BU
              </Button>
            </Col>,
          ],
          "className": "mb-4",
          "tag": "div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <h2>
                Business Unit List
              </h2>,
              "sm": 8,
              "tag": "div",
              "widths": Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Business Unit List",
              },
              "ref": null,
              "rendered": "Business Unit List",
              "type": "h2",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Button
                color="primary"
                tag={[Function]}
                to="/addBU"
              >
                Create BU
              </Button>,
              "className": "text-right",
              "sm": 4,
              "tag": "div",
              "widths": Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Create BU",
                "color": "primary",
                "tag": [Function],
                "to": "/addBU",
              },
              "ref": null,
              "rendered": "Create BU",
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "A",
        "nodeType": "class",
        "props": Object {
          "columns": Array [
            Object {
              "Header": "Channel Partner Id",
              "accessor": "channelPartnerId",
            },
            Object {
              "Header": "Channel Partner Name",
              "accessor": "channelPartnerName",
            },
            Object {
              "Header": "Channel Partner Type",
              "accessor": "channelPartnerType",
            },
            Object {
              "Header": "Currency",
              "accessor": "currency",
            },
            Object {
              "Cell": [Function],
              "Header": "Actions",
              "accessor": "[row identifier to be passed to button]",
              "filterable": false,
            },
          ],
          "data": Array [],
          "loading": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Row
            className="mb-4"
            tag="div"
          >
            <Col
              sm={8}
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <h2>
                Business Unit List
              </h2>
            </Col>
            <Col
              className="text-right"
              sm={4}
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <Button
                color="primary"
                tag={[Function]}
                to="/addBU"
              >
                Create BU
              </Button>
            </Col>
          </Row>,
          <Connect(ETable)
            columns={
              Array [
                Object {
                  "Header": "Channel Partner Id",
                  "accessor": "channelPartnerId",
                },
                Object {
                  "Header": "Channel Partner Name",
                  "accessor": "channelPartnerName",
                },
                Object {
                  "Header": "Channel Partner Type",
                  "accessor": "channelPartnerType",
                },
                Object {
                  "Header": "Currency",
                  "accessor": "currency",
                },
                Object {
                  "Cell": [Function],
                  "Header": "Actions",
                  "accessor": "[row identifier to be passed to button]",
                  "filterable": false,
                },
              ]
            }
            data={Array []}
            loading={true}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Col
                sm={8}
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <h2>
                  Business Unit List
                </h2>
              </Col>,
              <Col
                className="text-right"
                sm={4}
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <Button
                  color="primary"
                  tag={[Function]}
                  to="/addBU"
                >
                  Create BU
                </Button>
              </Col>,
            ],
            "className": "mb-4",
            "tag": "div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <h2>
                  Business Unit List
                </h2>,
                "sm": 8,
                "tag": "div",
                "widths": Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Business Unit List",
                },
                "ref": null,
                "rendered": "Business Unit List",
                "type": "h2",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <Button
                  color="primary"
                  tag={[Function]}
                  to="/addBU"
                >
                  Create BU
                </Button>,
                "className": "text-right",
                "sm": 4,
                "tag": "div",
                "widths": Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Create BU",
                  "color": "primary",
                  "tag": [Function],
                  "to": "/addBU",
                },
                "ref": null,
                "rendered": "Create BU",
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "A",
          "nodeType": "class",
          "props": Object {
            "columns": Array [
              Object {
                "Header": "Channel Partner Id",
                "accessor": "channelPartnerId",
              },
              Object {
                "Header": "Channel Partner Name",
                "accessor": "channelPartnerName",
              },
              Object {
                "Header": "Channel Partner Type",
                "accessor": "channelPartnerType",
              },
              Object {
                "Header": "Currency",
                "accessor": "currency",
              },
              Object {
                "Cell": [Function],
                "Header": "Actions",
                "accessor": "[row identifier to be passed to button]",
                "filterable": false,
              },
            ],
            "data": Array [],
            "loading": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
